src/parse.rs

```diff
@@ -58,7 +58,10 @@ macro_rules! impl_fromhex_array {
                     }
                     Ok(ret)
                 } else {
-                    Err(HexToArrayError::InvalidLength(2 * $len, 2 * iter.len()))
+                    Err(HexToArrayError::InvalidLength(InvalidLengthError {
+                        expected: 2 * $len,
+                        got: 2 * iter.len(),
+                    }))
```

Comment:
Possibly simplify it by omitting `HexToArrayError::InvalidLength` and calling `into` instead. (this is probably a matter of taste, I personally use `.into()` but don't care much)